python/cudf/dataframe/categorical.py

```diff
@@ -175,22 +175,16 @@ def to_arrow(self):
         if self.has_null_mask:
             # Necessary because PyArrow doesn't support from_buffers for
             # DictionaryArray yet
-            mask = pa.array(
-                # Why does expand_mask_bits return as int32?
-                cudautils.expand_mask_bits(
-                    len(self),
-                    self.nullmask.mem
-                )
-                .copy_to_host()
-                .astype(self.data.dtype)
-            )
-        indices = pa.array(self.cat().codes.data.mem.copy_to_host())
+            mask = cudautils.expand_mask_bits_logical_not(
+                len(self),
+                self.nullmask.mem
+            ).copy_to_host().astype('bool')
+        indices = pa.array(self.cat().codes.data.mem.copy_to_host(), mask=mask)
```

Comment:
There's a way to construct the pyarrow array from buffers instead of using the `mask` which won't require us to expand and flip the bits. See: https://github.com/rapidsai/cudf/blob/branch-0.7/python/cudf/dataframe/numerical.py#L202 for example.